### PR TITLE
Improved/fixed failed request queue

### PR DESCRIPTION
**Pull request by ErikBjare**
Completely forgot to file a PR for this, I've been running it for quite a while now.

### CODE REVIEW

**Comment by johan-bjareholt** [comment]
Merge needed

**Comment by johan-bjareholt** [comment]
How long have you tested this branch?

**Comment by ErikBjare** [comment]
I don't know, at least for 21 days I assume since that's when the first commits are from.

**Comment by ErikBjare** [comment]
But not after the merge with master...

**Comment by johan-bjareholt** [comment]
I don't think that the merge with master should be an issue. Would be nice though if you could extra check that most things in aw-webui look good though even after shutting aw-server on and off a few times.

**Comment by ErikBjare** [comment]
@johan-bjareholt Can we merge this now? master is unstable atm...

File: aw_client/client.py [hunk 1]
```diff
@@ -129,65 +127,110 @@ def disconnect(self):
...
-        with open(self.client.queue_file, "w") as queue_fp:
-            for request in self.queue.queue:
-                queue_fp.write(json.dumps(request) + "\n")
-
+        with open(self.queue_file, "w") as queue_fp:
+            while not self._queue.empty():
+                # The `block=False` and `if request is not None` stuff here is actually required, see Python docs.
+                request = self._queue.get(block=False)
+                if request is not None:
+                    queue_fp.write(json.dumps(request) + "\n")
```
Comment (ErikBjare) [inline aw_client/client.py]: Pretty sure this fixes #10 if not already fixed.

File: aw_client/client.py [hunk 2]
```diff
@@ -129,65 +127,110 @@ def disconnect(self):
...
+
+    def _try_connect(self) -> bool:
+        try:  # Try to connect
+            self.client._create_buckets()
+            return True
+        except req.RequestException:
+            return False
+
+    @RestartOnException
```
Comment (ErikBjare) [inline aw_client/client.py]: This should fix #8 acceptably.

File: aw_client/client.py [hunk 3]
```diff
@@ -129,65 +127,110 @@ def disconnect(self):
...
+        self._queue = Queue()
+
+        # Setup failed queues file
+        data_dir = get_data_dir("aw-client")
+        failed_queues_dir = os.path.join(data_dir, "failed_requests")
+        if not os.path.exists(failed_queues_dir):
+            os.makedirs(failed_queues_dir)
+        self.queue_file = os.path.join(failed_queues_dir, self.client.client_name)
+
         self._load_queue()
```
Comment (johan-bjareholt) [inline aw_client/client.py]: This line should be completely removed to fix clearing the file queue when not intended. My mistake

**Pull request merged**